### PR TITLE
ticdc: add ddl replication documents

### PR DESCRIPTION
**Pull request by ran-huang**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/14835
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: ticdc/ticdc-ddl.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+- rebase auto id
+- alter table index visibility
+- exchange partition
+- reorganize partition
+- alter table ttl
+- alter table remove ttl
+
+## DDL replication considerations
+
+Due to the lack of some context during the replication process, TiCDC has some constraints on the replication of DDLs of the `RENAME TABLE` type.
```
Comment (qiancai) [inline ticdc/ticdc-ddl.md:49]:
```suggestion
Due to the lack of some context during the replication process, TiCDC has some constraints on the replication of `RENAME TABLE` DDLs.
```

File: ticdc/ticdc-ddl.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+```
+
+TiCDC processes this type of DDL as follows:
+
+| DDL | Whether to replicate | Reason for the handling |
+| --- | --- | --- |
+| `RENAME TABLE test.t1 TO test.t2` | Replicate | `test.t1` matches the filter rule |
+| `RENAME TABLE test.t1 TO ignore.t1` | Replicate | `test.t1` matches the filter rule |
+| `RENAME TABLE ignore.t1 TO ignore.t2` | Ignore | `ignore.t1` does not match the filter rule |
+| `RENAME TABLE test.n1 TO test.t1` | Report an error and exit the replication | `test.n1` does not match the filter rule, but `test.t1` matches the filter rule. This operation is illegal. Please refer to the error message for handling. |
```
Comment (qiancai) [inline ticdc/ticdc-ddl.md:69]:
```suggestion
| `RENAME TABLE test.n1 TO test.t1` | Report an error and exit the replication | `test.n1` does not match the filter rule, but `test.t1` matches the filter rule. This operation is illegal. In this case, refer to the error message for handling. |
```

File: ticdc/ticdc-ddl.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+[filter]
+rules = ['test.t*']
+```
+
+TiCDC processes this type of DDL as follows:
+
+| DDL | Whether to replicate | Reason for the handling |
+| --- | --- | --- |
+| `RENAME TABLE test.t1 TO test.t2, test.t3 TO test.t4` | Replicate | All database names and table names match the filter rule. |
+| `RENAME TABLE test.t1 TO test.ignore1, test.t3 TO test.ignore2` | Replicate | The old database name, the old table name, and the new database name match the filter rule. |
```
Comment (qiancai) [inline ticdc/ticdc-ddl.md:90]:
```suggestion
| `RENAME TABLE test.t1 TO test.ignore1, test.t3 TO test.ignore2` | Replicate | The old database name, the old table names, and the new database name match the filter rule. |
```

File: ticdc/ticdc-ddl.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+| `RENAME TABLE test.t1 TO ignore.t1, test.t2 TO test.t22;` | Report an error | The new database name `ignore` does not match the filter rule. |
+| `RENAME TABLE test.t1 TO test1.t4, test.t3 TO test.t1, test.t4 TO test.t3;` | Report an error | The `RENAME TABLE` DDL swaps the names of `test.t1` and `test.t3` in one DDL statement, which TiCDC cannot handle correctly. Please refer to the error message for handling. |
```
Comment (qiancai) [inline ticdc/ticdc-ddl.md:92]: 如果要实现交换，这里的 test1.t4 要改成 test.t4 吗？

File: ticdc/ticdc-ddl.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+| --- | --- | --- |
+| `RENAME TABLE test.t1 TO test.t2` | Replicate | `test.t1` matches the filter rule |
+| `RENAME TABLE test.t1 TO ignore.t1` | Replicate | `test.t1` matches the filter rule |
+| `RENAME TABLE ignore.t1 TO ignore.t2` | Ignore | `ignore.t1` does not match the filter rule |
+| `RENAME TABLE test.n1 TO test.t1` | Report an error and exit the replication | `test.n1` does not match the filter rule, but `test.t1` matches the filter rule. This operation is illegal. Please refer to the error message for handling. |
+| rename table ignore.t1 to test.t1 | Report an error and exit the replication | Same reason as above. |
+
+### Rename multiple tables in a DDL statement
+
+If a DDL statement renames multiple tables, TiCDC only replicates the DDL statement when the old database name, old table name, and the new database name all match the filter rule.
```
Comment (qiancai) [inline ticdc/ticdc-ddl.md:74]:
```suggestion
If a DDL statement renames multiple tables, TiCDC only replicates the DDL statement when the old database name, old table names, and the new database name all match the filter rule.
```

File: ticdc/ticdc-ddl.md (hunk 4 above)
Comment (asddongmen) [reply ticdc/ticdc-ddl.md:92]: 不是。 TiCDC 无法同步交换表名的 DDL 语句，如果要实现交换，需要分成多条 DDL 执行。

Comment (qiancai) [reply ticdc/ticdc-ddl.md:92]: 对。只看前面两句呢， test1.t4 要改成 test.t4
image

File: ticdc/ticdc-ddl.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+```
+
+TiCDC processes this type of DDL as follows:
+
+| DDL | Whether to replicate | Reason for the handling |
+| --- | --- | --- |
+| `RENAME TABLE test.t1 TO test.t2, test.t3 TO test.t4` | Replicate | All database names and table names match the filter rule. |
+| `RENAME TABLE test.t1 TO test.ignore1, test.t3 TO test.ignore2` | Replicate | The old database name, the old table names, and the new database name match the filter rule. |
+| `RENAME TABLE test.t1 TO ignore.t1, test.t2 TO test.t22;` | Report an error | The new database name `ignore` does not match the filter rule. |
+| `RENAME TABLE test.t1 TO test1.t4, test.t3 TO test.t1, test.t4 TO test.t3;` | Report an error | The `RENAME TABLE` DDL swaps the names of `test.t1` and `test.t3` in one DDL statement, which TiCDC cannot handle correctly. Please refer to the error message for handling. |
```
Comment (qiancai) [inline ticdc/ticdc-ddl.md:92]:
```suggestion
| `RENAME TABLE test.t1 TO test.t4, test.t3 TO test.t1, test.t4 TO test.t3;` | Report an error | The `RENAME TABLE` DDL swaps the names of `test.t1` and `test.t3` in one DDL statement, which TiCDC cannot handle correctly. In this case, refer to the error message for handling. |
```

**Review by qiancai — APPROVED** [review]

File: ticdc/ticdc-ddl.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+TiCDC processes this type of DDL as follows:
+
+| DDL | Whether to replicate | Reason for the handling |
+| --- | --- | --- |
+| `RENAME TABLE test.t1 TO test.t2` | Replicate | `test.t1` matches the filter rule |
+| `RENAME TABLE test.t1 TO ignore.t1` | Replicate | `test.t1` matches the filter rule |
+| `RENAME TABLE ignore.t1 TO ignore.t2` | Ignore | `ignore.t1` does not match the filter rule |
+| `RENAME TABLE test.n1 TO test.t1` | Report an error and exit the replication | `test.n1` does not match the filter rule, but `test.t1` matches the filter rule. This operation is illegal. In this case, refer to the error message for handling. |
+| rename table ignore.t1 to test.t1 | Report an error and exit the replication | Same reason as above. |
```
Comment (ran-huang) [inline ticdc/ticdc-ddl.md:70]:
```suggestion
| `RENAME TABLE ignore.t1 TO test.t1` | Report an error and exit the replication | Same reason as above. |
```

**Pull request merged**